### ISSUE TITLE
Add Privacy Section for a User's Privacy Settings

### ISSUE THREAD

**Issue by gwwar** [Automattic/wp-calypso#22632]
To comply with GDPR we'll want to add a privacy settings section to allow a user to opt-out of tracking/advertising/cookies that we set on WordPress.com and Jetpack sites. This task is for the UI display work only.

I think a good candidate for this section would be under https://wordpress.com/me with a new section called "Privacy"

<img width="274" alt="screen shot 2018-02-20 at 12 03 31 pm" src="https://user-images.githubusercontent.com/1270189/36446464-2ce4f2f2-1636-11e8-8623-838d4415ee10.png">

cc @Automattic/lannister

**Comment by gwwar** [comment]
See https://github.com/Automattic/wp-calypso/pull/21652

**Comment by allendav** [comment]
Suggestion: Have this setting work be able to work in concert with a per-site setting, i.e. "Allow me to override this setting on a site-by-site basis" (that per-site setting could be a separate PR)

Rationale: Some users might be OK with WordPress.com tracking on some sites (e.g. thebloggess.com) but not others (e.g. bluehackers.org).

**Comment by gwwar** [comment]
@bperson @pesieminski are we thinking of storing settings with for a user on a per site basis? It might be a good enhancement after we meet other GDPR needs.

**Comment by bperson** [comment]
First try in #22648

**Comment by gwwar** [comment]
@bperson Do you mind linking your next related PRs here? It'll make it easier to track

**Comment by bperson** [comment]
PRs up for review:
1- #22693 introduces the `formBase` Hoc to save the status of the opt-out toggle in subsequent PRs
2- #21652 adds a redux action to inform `w.js` that it needs to drop ( or not ) tracking pixels.

Waiting on 1- and 2- before pushing those:
3- a PR to create the toggle in the privacy tab and uses the `formBase` HoC to persist the toggle value.
4- a PR to finish wiring up ( probably with a renderless component ): if we detect a change in the user settings, trigger the Redux action introduced in #21652 

And we should be good for the Tracks opt-out on the Calypso side of thing.

**Comment by bperson** [comment]
**Update time**

> 1- #22693 introduces the formBase Hoc to save the status of the opt-out toggle in subsequent PRs

#22693 is currently on standby following [this comment](https://github.com/Automattic/wp-calypso/pull/22693#issuecomment-369236635).

> 2- #21652 adds a redux action to inform w.js that it needs to drop ( or not ) tracking pixels.

#21652 is merged

> 3- a PR to create the toggle in the privacy tab and uses the formBase HoC to persist the toggle value.

#23231 is here! It uses the `formBase` mixin but it does the job. It has a little spin-off bugfix in #23232

> 4- a PR to finish wiring up ( probably with a renderless component ): if we detect a change in the user settings, trigger the Redux action introduced in #21652

still in the pipes

**Comment by bperson** [comment]
> 4- a PR to finish wiring up ( probably with a renderless component ): if we detect a change in the user settings, trigger the Redux action introduced in #21652

> still in the pipes

Attempt at getting this in #23256 

I think that's all of it for what I had in mind for the opt-out in Calypso :) I choose to pile up the last 3 PRs so that #23232 and #23256 can be "visually" tested via Privacy Settings but technically all of them can be reviewed independently and ( if un-piled ) could be merged in any order.

**Comment by kwight** [comment]
I think we'll want a SectionHeader for this (even if we only have one section for now); might look a little less empty (like on the My Profile page).

**Comment by bperson** [comment]
Alright closing this, I don't think it will get more love than what there is right now, we can always reopen it if there is another privacy push, maybe for CalGDPR in 2019?
